libdebug/debugger/internal_debugger.py

```diff
@@ -1901,6 +1938,253 @@ def set_stopped(self: InternalDebugger) -> None:
+            and self.handled_syscalls[syscall_number].on_exit_user is None
+        ):
+            if not self._is_in_background():
+                self.__polling_thread_command_queue.put((self.__threaded_cont_to_syscall, (thread,)))
+                self.__polling_thread_command_queue.put((self.__threaded_wait, ()))
+            else:
+                self.__threaded_cont_to_syscall(thread)
+                self.__threaded_wait()
+
+        if not self._is_in_background():
+            self._join_and_check_status()
+
+
+        self.debugging_interface.status_handler.executing_arbitrary_syscall = False
+
+        # Get return value.
+        retval = thread.syscall_return
+
+        thread.syscall_number = syscall_number
+
+        # Restore the original code.
+        try:
+            thread.memory[patch_ip, len_patch, "absolute"] = backup_code
+        except RuntimeError as e:
+            raise RuntimeError(
+                "Failed to restore the original instruction after syscall invocation. "
+                "The process likely died during the syscall invocation.",
+            ) from e
+
+        # Restore the registers
+
+        if not self._is_in_background():
+            self.__polling_thread_command_queue.put(
+                (self.__threaded_quick_regs_restore, (thread,)),
+            )
+            self._join_and_check_status()
+        else:
+            self.__threaded_quick_regs_restore(thread)
+
+        syscall_name = (
+            syscall_identifier
+            if isinstance(syscall_identifier, str)
+            else resolve_syscall_name(self.arch, syscall_number)
+        )
+
+        is_cloning_event = syscall_name in ["fork", "vfork", "clone", "clone3"]
+
+        if is_cloning_event:
+            # Complex handling of struct, maybe future work, but for an arbitrary call likely not necessary
+            if syscall_name == "clone3":
+                liblog.warning(
+                    "clone3 syscall invocation is experimental, check state of the new process / thread manually or use the base clone syscall"
+                )
+                return retval
+
+            CLONE_VM = 0x00000100
+            CLONE_THREAD = 0x00010000
+
+            thread_flags = effective_invocation_args[0] if len(effective_invocation_args) > 0 else 0
+
+            compatible_thread_flags = (
+                thread_flags & CLONE_VM == CLONE_VM
+                ) or (
+                thread_flags & CLONE_THREAD == CLONE_THREAD
+            )
+
+            is_process = syscall_name in ["fork", "vfork"] or not compatible_thread_flags
+
+            if is_process:
+                # If the syscall is a fork, we need to fix the state of the new process
+                new_pid = retval
+
+                reg_bit_count = get_platform_gp_register_size(self.arch) * 8
+                negative_threshold = 2 ** (reg_bit_count - 1)
+
+                if new_pid >= negative_threshold:
```

Comment:
`pid_t` is a signed 32 bit integer on all platforms, you could just check that new_pid.bit_length() <= 31, and it's not a platform-dependent check